src/webui/www/private/scripts/dynamicTable.js

```diff
@@ -813,6 +828,46 @@ window.qBittorrent.DynamicTable = (function() {
         getRowIds: function() {
             return this.rows.getKeys();
         },
+
+        selectNextRow: function() {
+            const visibleRows = $(this.dynamicTableDivId).getElements("tbody tr").filter(e => e.getStyle("display") !== "none");
+            const selectedRowId = this.getSelectedRowId();
+
+            let selectedIndex = -1;
+            visibleRows.forEach(function(row, i) {
```

Comment:
If I understand correctly, you should have used this form of loop: `for (let i = 0; i < xx; ++i) { if (predicate) {selectedIndex = i; break;} }` otherwise you'll loop over unnecessary entries in `visibleRows`.

Or you should be using `findIndex()` but I'm not sure it is available here.

The same to other similar instances.

Reply:
I originally tried `findIndex`, but it's not available on mootools arrays ([docs](https://mootools.net/core/docs/1.2.6/Native/Array)). Will switch this to a for loop.